Destiny.ScrimTracker.Logic/Repositories/GuardianEfficiencyRepository.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Destiny.ScrimTracker.Logic.Adapters;
+using Destiny.ScrimTracker.Logic.Models;
+using Microsoft.EntityFrameworkCore.Internal;
+
+namespace Destiny.ScrimTracker.Logic.Repositories
+{
+    public interface IGuardianEfficiencyRepository
+    {
+        GuardianEfficiency GetGuardianEfficiency(string guardianId);
+        IEnumerable<double> GetGuardianEfficiencies(string guardianId);
+        IEnumerable<GuardianEfficiency> GetGuardianAverageEfficiencies(string guardianId);
+        void UpdateGuardianEfficiency(GuardianEfficiency guardianEfficiency);
+        void UpdateGuardianMatchEfficiency(IEnumerable<GuardianMatchResult> guardianMatchResults);
+        int DeleteGuardianEfficiencies(string guardianId);
+        IEnumerable<string> DeleteEfficienciesForMatch(string matchId);
+    }
+    
+    public class GuardianEfficiencyRepository : IGuardianEfficiencyRepository
+    {
+        private readonly DatabaseContext _databaseContext;
+
+        public GuardianEfficiencyRepository(DatabaseContext databaseContext)
+        {
+            _databaseContext = databaseContext;
+        }
+        
+        public GuardianEfficiency GetGuardianEfficiency(string guardianId)
+        {
+            var guardianEfficiency = _databaseContext.GuardianEfficiencies.Where(eff => eff.GuardianId == guardianId)
+                .OrderByDescending(eff => eff.TimeStamp).FirstOrDefault();
+
+            return guardianEfficiency;
+        }
+
+        public IEnumerable<double> GetGuardianEfficiencies(string guardianId)
+        {
+            return _databaseContext.GuardianMatchResults.Where(eff => eff.GuardianId == guardianId && eff.MatchId != null).Select(gmr => gmr.Efficiency);
+        }
+
+        public IEnumerable<GuardianEfficiency> GetGuardianAverageEfficiencies(string guardianId)
+        {
+            var efficiencies = _databaseContext.GuardianEfficiencies.Where(eff => eff.GuardianId == guardianId && eff.MatchId != null);
+
+            if (efficiencies == null || !efficiencies.Any())
+            {
+                return new GuardianEfficiency[] {};
+            }
+
+            return efficiencies;
+        }
+
+        public void UpdateGuardianEfficiency(GuardianEfficiency guardianEfficiency)
+        {
+            _databaseContext.Add(guardianEfficiency);
+            _databaseContext.SaveChanges();
```

Comment:
Might be a good idea to use all the async method in EF: SaveChangesAsync(), ToListAsync(), FirstOrDefaultAsync, etc.